utils/get_ci_error_statistics.py

```diff
@@ -0,0 +1,154 @@
+            redirect_url = line[len("< Location: ") :]
+            r = requests.get(redirect_url, allow_redirects=True)
+            p = os.path.join(output_dir, f"{artifact_name}.zip")
+            open(p, "wb").write(r.content)
+            break
+
+
+def get_errors_from_single_artifact(artifact_zip_path):
+    """Extract errors from a downloaded artifact (in .zip format)"""
+    errors = []
+    failed_tests = []
+
+    with zipfile.ZipFile(artifact_zip_path) as z:
+        for filename in z.namelist():
+            if not os.path.isdir(filename):
+                # read the file
+                if filename in ["failures_line.txt", "summary_short.txt"]:
+                    with z.open(filename) as f:
+                        for line in f:
+                            line = line.decode("UTF-8").strip()
+                            if filename == "failures_line.txt":
+                                try:
+                                    # `error_line` is the place where `error` occurs
+                                    error_line = line[: line.index(": ")]
+                                    error = line[line.index(": ") + len(": ") :]
+                                    errors.append([error_line, error])
+                                except Exception:
+                                    # skip un-related lines
+                                    pass
+                            elif filename == "summary_short.txt" and line.startswith("FAILED "):
+                                # `test` is the test method that failed
+                                test = line[len("FAILED ") :]
+                                failed_tests.append(test)
+
+    if len(errors) != len(failed_tests):
+        raise ValueError(
+            f"`errors` and `failed_tests` should have the same number of elements. Got {len(errors)} for `errors` "
+            f"and {len(failed_tests)} for `failed_tests` instead. The test reports in {artifact_zip_path} have some"
+            " problem."
+        )
+
+    return errors, failed_tests
+
+
+def get_all_errors(artifact_dir):
+    """Extract errors from all artifact files"""
+
+    errors = []
+    failed_tests = []
+
+    paths = [os.path.join(artifact_dir, p) for p in os.listdir(artifact_dir) if p.endswith(".zip")]
+
+    for p in paths:
+        _errors, _failed_tests = get_errors_from_single_artifact(p)
+        errors.extend(_errors)
+        failed_tests.extend(_failed_tests)
+
+    return errors, failed_tests
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    # Required parameters
+    parser.add_argument(
+        "--workflow_run_id", default=None, type=str, required=True, help="A GitHub Actions workflow run id."
+    )
+    parser.add_argument(
+        "--output_dir",
+        default=None,
+        type=str,
+        required=True,
+        help="Where to store the downloaded artifacts and other result files.",
+    )
+    parser.add_argument(
```

Comment:
I don't like this place much for the security concern. But at this moment, I might be the only person.